File: MG/App/Frontend/moj_grad_app/lib/models/modelsViews/commentInfo.dart
class CommentInfo {
  int id;
  int postId;
  int userId;

  String content;
  String username;
  int commentLikes;
  int commentDislikes;
  int isLiked;
  String profilePhoto;

  bool get isLike {
    if (isLiked == 1)
      return true;
    else
      return false;
  }

  bool get isDislike {
    if (isLiked == -1)
      return true;
    else
      return false;
  }

  CommentInfo(id, postId, userId, content, username, commentLikes,
      commentDislikes, isLiked, profilePhoto) {
    this.id = id;
    this.postId = postId;
    this.userId = userId;
    this.content = content;
    this.username = username;
    this.commentLikes = commentLikes;
    this.commentDislikes = commentDislikes;
    this.isLiked = isLiked;
    this.profilePhoto = profilePhoto;
  }

  Map<String, dynamic> toMap() {
    var map = Map<String, dynamic>();
    if (id != null) map['id'] = id;
    map['postId'] = postId;
    map['userId'] = userId;
    map['content'] = content;
    map['username'] = username;
    map['commentLikes'] = commentLikes;
    map['commentDislikes'] = commentDislikes;
    map['isLiked'] = isLiked;
    map['profilePhoto'] = profilePhoto;
    return map;
  }

  CommentInfo.fromObject(dynamic o) {
    this.id = o['id'];
    this.postId = o['postId'];
    this.userId = o['userID'];
    this.content = o['content'];
    this.username = o['username'];
    this.commentLikes = o['commentLikes'];
    this.commentDislikes = o['commentDislikes'];
    this.isLiked = o['isLiked'];
    this.profilePhoto = o['profilePhoto'];
  }
}

File: MG/App/Backend/mojGradApp/mojGradApp/Models/InfoModels/CommentInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace mojGradApp.Models.InfoModels
{
    public class CommentInfo
    {
        public int Id { get; set; }
        public int PostID { get; set; }
        public int UserID { get; set; }
        public string Content { get; set; }
        public DateTime Time { get; set; }

        public string Username { get; set; }
        public int CommentLikes { get; set; }
        public int CommentDislikes { get; set; }
        public int IsLiked { get; set; }
        public string ProfilePhoto { get; set; }

        public CommentInfo(Comments comment)
        {
            this.Id = comment.Id;
            this.PostID = comment.PostID;
            this.UserID = comment.UserID;
            this.Content = comment.Content;
            this.Time = comment.Time;
            this.Username = comment.User.Username;
            this.CommentLikes = comment.CommentReaction.Count(x => x.CommentID == this.Id && x.Type == 1);
            this.CommentDislikes = comment.CommentReaction.Count(x => x.CommentID == this.Id && x.Type == -1);
            this.ProfilePhoto = comment.User.ProfilePhotoPath;
        }

        public CommentInfo(Comments comment, int userID) : this(comment)
        {
            if (comment.CommentReaction.Where(x => x.UserID == userID && x.Type == -1).FirstOrDefault() != null)
                this.IsLiked = -1;
            else if (comment.CommentReaction.Where(x => x.UserID == userID && x.Type == 1).FirstOrDefault() != null)
                this.IsLiked = 1;
            else
                this.IsLiked = 0;
        }
    }
}
